ext/rotoscope/stack.c

```diff
@@ -0,0 +1,104 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "stack.h"
+
+rs_stack_frame_t root_context;
+
+bool stack_full(rs_stack_t *stack)
+{
+  return stack->top >= stack->capacity - 1;
+}
+
+bool stack_empty(rs_stack_t *stack)
+{
+  return stack->top < 0;
+}
+
+static void resize_buffer(rs_stack_t *stack)
+{
+  unsigned int newsize = stack->capacity * 2;
+  rs_stack_frame_t *new_contents = (rs_stack_frame_t *)malloc(sizeof(rs_stack_frame_t) * newsize);
+  memcpy(new_contents, stack->contents, sizeof(rs_stack_frame_t) * stack->capacity);
+  stack->capacity = newsize;
+  stack->contents = new_contents;
```

Comment:
🤦‍♂️ I'd like to blame Hack Days for this one but I think I just completely overlooked it. Good catch.